src/expression.ts

```diff
@@ -98,20 +101,27 @@ export class ElementAccessExpr extends BaseNode<"ElementAccessExpr"> {
   }
 }
 
+export const isArgumentExpr = typeGuard("ArgumentExpr");
+
+export class ArgumentExpr extends BaseNode<"ArgumentExpr"> {
```

Comment:
Is it `Argument` or `ArgumentExpr`. I only ask because I don't think it's technically an expression. The node can't be used where an expression can be. Eg `ReturnStmt(ArgumentExpr)`. It should be renamed.

Reply:
Ok, it can just be a an interface and not a `Expression` class. Though I think we use expression loosely. There are a few applications of the `PropAssignmentExpr` and `SpreadAssignExpr`, you cannot `return ...identifier`.

Reply:
It can just be a class that extends BaseNode. Not an interface.

Reply:
There is no `ts.Argument`, just an array of arguments. Attaching a name to the argument was introduced by Functionless.

Reply:
```ts
readonly arguments: NodeArray<Expression>;
```

Reply:
Hmmmmmmm ok. Ok let's call it an expression. Sorry for the confusion.

Reply:
I think I was wrong. Getting confused